src/software/util/sml_fsm/sml_fsm.h

```diff
@@ -3,9 +3,160 @@
 #include <functional>
 #include <include/boost/sml.hpp>
 #include <queue>
+#include <string>
+#include <map>
+#include "proto/play_info_msg.pb.h"
+#include "software/logger/logger.h"
 
 #include "software/util/typename/typename.h"
 
+/**
+ * This class declares a logger to log the FSM State Changes
+ * https://boost-ext.github.io/sml/examples.html#logging
+ *
+ * This logger DOES NOT FOLLOW the Singleton design pattern.
+ * This is because to maintain synchronization between AI logs and actual AI actions, the controlling Play and each
+ * of its tactics need to have their own separate loggers.
+ */
+
+class FSMLogger {
+public:
+    /**
+     * Constructors for FSMLogger
+     *
+     * @param robot_id the Robot ID of the Tactic that this logger is attached to.
+     *                  (nullopt if this logger is attached to a play)
+     */
+    explicit FSMLogger(std::optional<unsigned int> robot_id) : robot_id(robot_id) {}
```

Comment:
Consider having two constructors (another one for `Play` that doesn't take any arguments)